test/integration/targets/mysql_db/tasks/multi_db_create_delete.yml

```
@@ -0,0 +1,699 @@
+
+- name: run command to list databases like specified database name
+  command: mysql "-e show databases like 'database%';"
+  register: mysql_result
+
+- name: assert that databases does exist (since check mode)
+  assert:
+    that:
+      - "'{{ db1_name }}' in mysql_result.stdout"
+      - "'{{ db2_name }}' in mysql_result.stdout"
+      - "'{{ db3_name }}' in mysql_result.stdout"
+      - "'{{ db4_name }}' not in mysql_result.stdout"
+      - "'{{ db5_name }}' not in mysql_result.stdout"
+
+- name: state dump - file name should not exist (since prior dump operation performed via check mode)
+  file: name={{ dump1_file }} state=absent
+
+# ==========================================================================
+# Dump non-existing databases (check mode)
+- name: Dump non-existing databases (check mode)
+  mysql_db:
+    name:
+      - "{{ db4_name }}"
+      - "{{ db5_name }}"
+    state: dump
+    target: "{{ dump1_file }}"
+    login_unix_socket: '{{ mysql_socket }}'
+  register: check_mode_dump_result
+  ignore_errors: True
+  check_mode: yes
+
+- name: assert successful completion of dump operation using check mode
+  assert:
+    that:
+       - "'Cannot dump database' in check_mode_dump_result['msg']"
+
+- name: run command to list databases like specified database name
+  command: mysql "-e show databases like 'database%';"
+  register: mysql_result
+
+- name: assert that databases does exist (since delete via check mode)
+  assert:
+    that:
+      - "'{{ db1_name }}' in mysql_result.stdout"
+      - "'{{ db2_name }}' in mysql_result.stdout"
+      - "'{{ db3_name }}' in mysql_result.stdout"
+      - "'{{ db4_name }}' not in mysql_result.stdout"
+      - "'{{ db5_name }}' not in mysql_result.stdout"
+
+- name: state dump - file name should not exist (since prior dump operation performed via check mode)
+  file: name={{ dump1_file }} state=absent
+
+# ==========================================================================
+# Dump existing databases
+- name: Dump existing databases
+  mysql_db:
+    name:
+      - '{{ db1_name }}'
+      - '{{ db2_name }}'
+      - '{{ db3_name }}'
+    state: dump
+    target: '{{ dump1_file }}'
+    login_unix_socket: '{{ mysql_socket }}'
+  register: dump_result
+
+- name: assert successful completion of dump operation
+  assert:
+    that:
+       - dump_result.changed == true
+
+- name: run command to list databases like specified database name
+  command: mysql "-e show databases like 'database%';"
+  register: mysql_result
+
+- name: assert that databases does exist
+  assert:
+    that:
+      - "'{{ db1_name }}' in mysql_result.stdout"
+      - "'{{ db2_name }}' in mysql_result.stdout"
+      - "'{{ db3_name }}' in mysql_result.stdout"
+      - "'{{ db4_name }}' not in mysql_result.stdout"
+      - "'{{ db5_name }}' not in mysql_result.stdout"
+
+- name: state dump - file name should exist
+  file: name={{ dump1_file }} state=file
```

```suggestion
  file: name={{ dump1_file }} state=file
```
we could also grep the dump file by shell module to be sure that there are something like CREATE DATABASE blabla

---

fixed